### PR TITLE
Make the 'add_res_data()' function use res_grp_name instead of lookup()

### PR DESCRIPTION
**Pull request by 14rcole**
`lookup('vars', 'role_name')` confuses things.  This makes it easier for people developing roles to understand

Signed-off-by: Ryan Cole <ryan@localhost.localdomain>

### CODE REVIEW

**Comment by JunqiZhang0** [comment]
@14rcole Could you get rid of the azure stuff so that we don't have conflicts? Thanks!

**Comment by 14rcole** [comment]
What do you mean?  Not change azure so we can avoid rebasing?

**Comment by JunqiZhang0** [comment]
Yes, I worry rebase maybe mess up some of my code

**Comment by 14rcole** [comment]
I don't think that's a reason to do a partial fix.  We shouldn't split a fix across multiple PRs just to avoid rebasing, that's why its there

**Comment by 14rcole** [comment]
[test]

**Comment by 14rcole** [comment]
[test]

**Comment by 14rcole** [comment]
[test]

**Comment by 14rcole** [comment]
[test]

**Comment by 14rcole** [comment]
[test]

**Comment by JunqiZhang0** [comment]
[test]

**Comment by 14rcole** [comment]
[test]

**Comment by JunqiZhang0** [comment]
still failing on unit tests

**Comment by 14rcole** [comment]
[test]

**Comment by 14rcole** [comment]
[test]

**Comment by 14rcole** [comment]
[test]

File: linchpin/provision/roles/azure/tasks/provision_azure_vm.yml [hunk 1]
```diff
@@ -99,4 +99,14 @@
   include: provisionVM.yml nameOfvm={{nameOfvm}}
   loop: "{{ name_list}}"
   loop_control:
-    loop_var: nameOfvm
\ No newline at end of file
+    loop_var: nameOfvm
```
Comment (JunqiZhang0) [inline linchpin/provision/roles/azure/tasks/provision_azure_vm.yml]: You don't need to add resource type here,  this is just a wrapper of VM provision, the real provision part is in provisionVM.yml

**Comment by 14rcole** [comment]
[test]

**Comment by 14rcole** [comment]
@JunqiZhang0 everything should work now, do you want to review it one more time?

**Comment by 14rcole** [comment]
[merge]

**Pull request merged**